airflow/cli/commands/legacy_commands.py

```diff
@@ -0,0 +1,57 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from argparse import ArgumentError
+
+COMMAND_MAP = {
+    # "worker": "celery worker",
+    # "flower": "celery flower",
+    "trigger_dag": "dags trigger",
+    "delete_dag": "dags delete",
+    "show_dag": "dags show",
+    "list_dag": "dags list",
+    "dag_status": "dags status",
+    "backfill": "dags backfill",
+    "list_dag_runs": "dags list_runs",
+    "pause": "dags pause",
+    "unpause": "dags unpause",
+    "test": "tasks test",
+    "clear": "tasks clear",
+    "list_tasks": "tasks list",
+    "task_failed_deps": "tasks failed_deps",
+    "task_state": "tasks state",
+    "run": "tasks run",
+    "render": "tasks render",
+    "initdb": "db init",
+    "resetdb": "db reset",
+    "upgradedb": "db upgrade",
+    "checkdb": "db check",
+    "shell": "db shell",
+    "pool": "pools",
+    "list_users": "users list",
+    "create_user": "users create",
+    "delete_user": "users delete"
+}
+
+
+def check_value(action, value):
+    """ Checks command value and raise error if value is in removed command """
```

Comment:
What you need to add here is this if (from _check_value):

```
    if action.choices is not None and value not in action.choices:
```

The problem is that the Default Help parser is run for both - the main "parser" and "celery" subcommand parser. And when it is run for the subcommand parser it should be accepted.

If you add the 'if' before, the check for old parameters will only be done if 'worker' is not valid choice.

Reply:
Thanks! That solved it!